src/main/MQ2ImGuiConsole.cpp

```diff
@@ -27,1504 +27,1638 @@
+			if (gImGuiConsole != nullptr)
+				gImGuiConsole->ClearLog();
 
-			if (ImGui::GetCurrentWindowRead()->Viewport->ID != mainViewport->ID)
-			{
-				// Activate the main viewport window.
-				::SetActiveWindow((HWND)mainViewport->PlatformHandle);
-			}
+			return;
 		}
-	}
-}
 
-void MQConsoleCommand(SPAWNINFO* pChar, char* Line)
-{
-	char szCommand[MAX_STRING] = { 0 };
-	GetArg(szCommand, Line, 1);
+		if (!_stricmp("toggle", szCommand))
+		{
+			gbToggleConsoleRequested = true;
+			return;
+		}
 
-	if (!_stricmp("clear", szCommand))
-	{
-		if (gImGuiConsole != nullptr)
-			gImGuiConsole->ClearLog();
+		if (!_stricmp("show", szCommand))
+		{
+			gbSetConsoleVisibilityRequest = true;
+			return;
+		}
 
-		return;
-	}
+		if (!_stricmp("hide", szCommand))
+		{
+			gbSetConsoleVisibilityRequest = false;
+			return;
+		}
 
-	if (!_stricmp("toggle", szCommand))
-	{
-		gbToggleConsoleRequested = true;
-		return;
+		WriteChatf("Usage: /mqconsole [command]");
+		WriteChatf("  Commands: clear, toggle, show, hide");
 	}
 
-	if (!_stricmp("show", szCommand))
+	int FontNumberFromString(std::string fontString)
 	{
-		gbSetConsoleVisibilityRequest = true;
-		return;
+		ImGuiIO& io = ImGui::GetIO();
+		for (int i = 0; i < io.Fonts->Fonts.Size; i++)
+		{
+			std::string fontstr = io.Fonts->Fonts[i]->GetDebugName();
+			if (strcmp(fontstr.c_str(), fontString.c_str()) == 0)
+			{
+				return i;
+			}
+		}
+		return 0;
 	}
 
-	if (!_stricmp("hide", szCommand))
+	static void ConsoleSettings()
 	{
-		gbSetConsoleVisibilityRequest = false;
-		return;
-	}
+		ImGuiIO& io = ImGui::GetIO();
+		if (ImGui::Checkbox("Show Console on Load", &s_consoleVisibleOnStartup))
+		{
+			WritePrivateProfileBool("MQConsole", "ShowMacroQuestConsole", s_consoleVisibleOnStartup, mq::internal_paths::MQini);
+		}
 
-	WriteChatf("Usage: /mqconsole [command]");
-	WriteChatf("  Commands: clear, toggle, show, hide");
-}
+		ImGui::SameLine();
+		mq::imgui::HelpMarker("This feature allows you to automatically show the MacroQuest Console upon load.");
 
-static void ConsoleSettings()
-{
-	if (ImGui::Checkbox("Show Console on Load", &s_consoleVisibleOnStartup))
-	{
-		WritePrivateProfileBool("MacroQuest", "ShowMacroQuestConsole", s_consoleVisibleOnStartup, mq::internal_paths::MQini);
-	}
+		if (ImGui::Checkbox("Lock Window", &s_isConsoleLocked))
+		{
+			WritePrivateProfileBool("MQConsole", "LockConsoleWindow", s_isConsoleLocked, mq::internal_paths::MQini);
+		}
+		ImGui::SameLine();
+		mq::imgui::HelpMarker("This feature allows you to automatically lock the MacroQuest Console.");
 
-	ImGui::SameLine();
-	mq::imgui::HelpMarker("This feature allows you to automatically show the MacroQuest Console upon load.");
+		if (ImGui::Checkbox("Hide Titlebar", &s_isTitleBarShown))
+		{
+			WritePrivateProfileBool("MQConsole", "NoTitlebar", s_isTitleBarShown, mq::internal_paths::MQini);
+		}
+		ImGui::SameLine();
+		mq::imgui::HelpMarker("This feature allows you to hide MacroQuest Menu Bar.");
```

Comment:
+the, Menu Bar should be lower cased. Should probably make some mention of the title bar since the option is called "Hide Titlebar"